File: examples/proof_of_work.js
#!/usr/bin/env node
var crypto = require("crypto");

var transactions = "Hello, world!";
var finished = false;
var nonce = 0;
var difficulty = 4;

while (!finished) {
  var hash = crypto.createHash("sha256");
  hash.update(transactions + nonce);
  var digest = hash.digest('hex');

  if (digest.slice(0, difficulty) == "0".repeat(difficulty)) {
    console.log("DONE");
    console.log(transactions + nonce);
    console.log(digest);
    finished = true;
  }

  nonce += 1;
}

File: examples/proof_of_work.rb
#!/usr/bin/env ruby
require 'digest'

finished = false
transactions = "Hello, world!"
nonce = 0
difficulty = 4

while !finished
  digest = Digest::SHA256.hexdigest(transactions + nonce.to_s)

  if digest[0, difficulty] == "0" * difficulty
    puts "DONE"
    puts transactions + nonce.to_s
    puts digest
    finished = true
  end

  nonce += 1
end


# imagine you are trying to a bunch of different keys
